### ISSUE TITLE
feature-request: Add flash messages, similar to rails

### ISSUE THREAD

**Issue by andrewzah** [kemalyst/kemalyst#9]
It would be nice for kemalyst to have something similar to Rails' [flash system](http://api.rubyonrails.org/classes/ActionDispatch/Flash.html).

So one could write:
`controller.cr`
```crystal
class Show < Kemalyst::Controller
  def call(context)
    id = context.params["id"]
    if post = Post.find id
      render "post/show.ecr", "posts.ecr"
    else
      flash[:error] = "Post ID #{id} not found"
      redirect "/posts"
    end
  end
end
```
`main.ecr`
```ecr
<div class="row">
  <% flash.each do |key, value| %>
    <div class="alert alert-<%= key %>">
      <p> <% value %> </p>
    </div>
  <% end %>
</div>
```

**Comment by drujensen** [comment]
@azah I have released support for flash messages in v0.2.3.  

The syntax is slightly different than rails because you cannot create symbols dynamically in crystal so the keys are Strings instead of Symbols: 
```crystal
context.flash["error"] = "Post ID #{id} not found"
```
The flash[] is a modified hash to track which keys have been read and mark them for removal.

In your template:
```crystal
<% context.flash.each do |key, value| %>
  <div class="alert alert-<%= key %>">
    <p> <%= value %> </p>
   </div>
<% end %>
```

**Comment by andrewzah** [comment]
Thanks!
